script/common/actor.js

```diff
@@ -45,7 +48,8 @@ export class DarkHeresyActor extends Actor {
     this.system.insanityBonus = Math.floor(this.insanity / 10);
     this.system.corruptionBonus = Math.floor(this.corruption / 10);
     this.psy.currentRating = this.psy.rating - this.psy.sustained;
-    this.initiative.bonus = this.characteristics[this.initiative.characteristic].bonus;
+    if (this.type == "starship") {this.initiative.bonus = this.shipdetection;}
```

Comment:
this should be ===. the == doesn't check the content its purely coincidence if it works as expected